job-engine/commons/src/main/java/org/eclipse/kapua/job/engine/commons/operation/DefaultTargetReader.java

```diff
@@ -112,14 +117,22 @@ public Object readItem() throws Exception {
         JobLogger jobLogger = jobContextWrapper.getJobLogger();
         jobLogger.setClassLog(LOG);
 
-        jobLogger.info("Reading item...");
+        JobTargetQuery query = jobTargetFactory.newQuery(jobContextWrapper.getScopeId());
+
+        AndPredicate andPredicate = query.andPredicate(
+            query.attributePredicate(JobTargetAttributes.JOB_ID, jobContextWrapper.getJobId())
+        );
+        query.setPredicate(andPredicate);
+        JobTargetListResult jobTargets = KapuaSecurityUtils.doPrivileged(() -> jobTargetService.query(query));
+
+        jobLogger.info("Reading target: {} (id: {})...", getTargetDisplayName(jobTargets.getItem(0)), jobTargets.getItem(0).getId().toCompactId());
```

Comment:
This also does not handle query with 0 Devices found and will produce `NullPointerException`